rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,179 @@
+
+1. The frontend uses ECS deployments and the API is well on the same track. This makes it possible for them to share some deployment code.
+
+1. To the RFC author, the API and frontend are very familiar so merging them would be easier. Adding a third component would make the task daunting.
+
+1. Merging incurs a productivity hit for the initial transition. So merging everything in one swoop is not ideal.
+
+1. The API’s comprehensive tooling for developer documentation can benefit frontend devs and create a unified docs site for contributors.
+
+1. The API is already organised by stack folders so the `frontend/` directory will fit right in with the others like `api/` and `ingestion_server/`.
+
+1. The API and frontend share identical tooling for Git hooks, linting and formatting. We will fight our tools less and encounter minimal friction.
+
+   - In fact, we employ a number of hacks to install and configure pre-commit for the frontend. Merging it with the API eliminates the need for such hacks.
+
+1. The entire system can be integration tested during releases. The real API, populated with test data, can even replace the Talkback server.
+
+The `WordPress/openverse-api` repo will absorb the `WordPress/openverse-frontend` repo. The `WordPress/openverse-catalog` will also be merged, _later_.
+
+### Reference
+
+I'm following the steps listed below in a fork at [@dhruvkb/monopenverse](https://github.com/dhruvkb/monopenverse/). You can refer to the fork, but note that it is a comes from a place of haste and has not been treated with the same level of love and care that the final treatment will receive.
+
+### Step 0: Prerequisites
+
+#### Get the timing right
+
+The first step will be to release the frontend, call a code freeze and pause work on it. This is to prevent the frontend repo from continuing to drift as we merge a snapshot of it with the API.
+
+This can prove difficult given how productive our team is, so we will need to channel this productivity towards the catalog in the meantime. I can foresee the end-to-end migration taking one week (ideal scenario) to one fortnight (worst case scenario).
+
+### Step 1: Merge with histories
+
+This is a quick process.
+
+1. Move the entire content of frontend inside a `frontend/` directory, except the following top-level files and folders. Please comment if you can add to this list.
+
+   - `.github/`
+   - `.editorconfig`
+   - `justfile`
+   - `.pre-commit-config.yaml`
+   - `prettier.config.js`
+   - `.prettierignore`
+   - `.eslintrc.js` (need to update references in nested `eslintrc.js` files)
+   - `.eslintignore`
+   - <s>`.gitignore`</s> (better to move it into the `frontend/` directory and update some absolute paths)
+
+1. Create the final commit on the `WordPress/openverse-frontend` repo. After the merge we might want to add a notice about the migration to the `README.md` file but GitHub's built-in archival process could suffice here.
```

Comment:
I think we _definitely_ want to add a README notice and use the GitHub archiving functionality together. The README information will crucially link to the new repository. We should also update the repo description with the new url as well, here:

![CleanShot 2022-12-02 at 13 20 13](https://user-images.githubusercontent.com/6351754/205359869-16d87694-9452-4b9c-b541-a07f57599ac5.png)